samtranslator/intrinsics/actions.py

```diff
@@ -506,9 +502,9 @@ def resolve_resource_id_refs(
         remaining = splits[1:]  # if any
 
         resolved_value = supported_resource_id_refs.get(logical_id)
-        return self._get_resolved_dictionary(input_dict, key, resolved_value, remaining)  # type: ignore[no-untyped-call]
+        return self._get_resolved_dictionary(input_dict, key, resolved_value, remaining)
 
-    def _check_input_value(self, value):  # type: ignore[no-untyped-def]
+    def _check_input_value(self, value: Any) -> bool:
```

Comment:
The comment says that `value` must be an array. Why `Any` then?

Reply:
I think the reason for the comment to say Value must be an array is because it does a type check in the function and if it's not list, we immediately return false.